src/scripts/main.js

```diff
@@ -1,3 +1,62 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const appendRow = document.querySelector('.append-row');
+  const removeRow = document.querySelector('.remove-row');
+  const appendColumn = document.querySelector('.append-column');
+  const removeColumn = document.querySelector('.remove-column');
+  const table = document.querySelector('.field');
+
+  const updateButtons = () => {
+    const allRows = table.rows.length;
+    const allCols = table.rows[0].cells.length;
+
+    appendRow.disabled = allRows >= 10;
+    removeRow.disabled = allRows <= 2;
+    appendColumn.disabled = allCols >= 10;
+    removeColumn.disabled = allCols <= 2;
+  };
```

Comment:
The `updateButtons` function is called multiple times throughout the code. It might be a good idea to consider debouncing this function to improve performance, especially if the number of rows or columns becomes large.